packages/config/src/validate/validations.js

```diff
@@ -216,6 +217,22 @@ const POST_NORMALIZE_VALIDATIONS = [
       functions: { [prevPath[1]]: { ignored_node_modules: ['module-one', 'module-two'] } },
     }),
   },
+  {
+    property: 'functions.*.included_files',
+    check: isArrayOfStrings,
+    message: 'must be an array of strings.',
+    example: (value, key, prevPath) => ({
+      functions: { [prevPath[1]]: { included_files: ['directory-one/file1', 'directory-two/**/*.jpg'] } },
+    }),
+  },
+  {
+    property: 'functions.*.included_files',
+    check: (value) => value.every(isInsideRoot),
```

Comment:
Would it make sense to separate validation for:
  - `property: 'functions.*.included_files'`: checking if it is an array
  - `property: 'functions.*.included_files.*`: checking if each item is a string and is inside root

Like this, if a file path is not inside root, the validation message will be more precise, showing only that element to users?

Reply:
I added two validations: the first checks if it's an array of strings, the second checks whether the paths are inside the root. I did this precisely to provide a more useful error message if a path is not inside the root.

The only difference from what you said was to move the string validation from the former to the latter, but I'm not sure I understand the benefits. Can you clarify?

Reply:
Breaking down between string validation and `isInsideRoot` sounds good.

What I was suggesting was also breaking down array validation (`functions.*.included_files`) vs array item validation (`functions.*.included_files.*`). We should double check but I believe validating each individual array item would allow us to pinpoint to users which specific item triggered the validation.

Reply:
Oh, I see! I had missed the additional `.*`! 🤦🏻  That makes a lot of sense, I'll update. Thanks!

Reply:
Done in https://github.com/netlify/build/pull/2681/commits/aa1ff4b7118ba02da28fdefd9eb2bc0c56f566ab.